### PR TITLE
Feature/[ios] Support discovery methods

### PR DESCRIPTION
**Pull request by TomoLV**

### CODE REVIEW

File: ios/Classes/FlutterBleLibPlugin.m [hunk 1]
```diff
@@ -152,14 +158,40 @@ - (void)logLevel:(FlutterMethodCall *)call result:(FlutterResult)result {
...
+}
+
+- (void)characteristics:(FlutterMethodCall *)call result:(FlutterResult)result {
+    [_manager servicesForDevice:call.arguments[ARGUMENT_KEY_DEVICE_IDENTIFIER]
+                        resolve:[self resolveForCharacteristics:result
+                                                    serviceUuid:call.arguments[ARGUMENT_KEY_SERVICE_UUID]]
+                         reject:[self rejectForFlutterResult:result]];
+}
+
+// MARK: - MBA Methods - Characteristics opertions
```
Comment (mikolak) [inline ios/Classes/FlutterBleLibPlugin.m]: Wrong PR

File: ios/Classes/FlutterBleLibPlugin.m [hunk 2]
```diff
@@ -152,14 +158,40 @@ - (void)logLevel:(FlutterMethodCall *)call result:(FlutterResult)result {
...
+
+- (void)readCharacteristicsForDevice:(FlutterMethodCall *)call result:(FlutterResult)result {
```
Comment (mikolak) [inline ios/Classes/FlutterBleLibPlugin.m]: Wrong PR

File: ios/Classes/FlutterBleLibPlugin.m [hunk 3]
```diff
@@ -178,9 +210,40 @@ - (void)dispatchEvent:(NSString * _Nonnull)name value:(id _Nonnull)value {
 
 // MARK: - Utility methods
 
-- (Resolve)jsonStringResolveForFlutterResult:(FlutterResult)result {
-    return ^(NSString *jsonString) {
-        result([JSONStringifier jsonStringFromJSONObject:jsonString]);
+- (Resolve)resolveForServicesForDevice:(FlutterResult)result {
+    return ^(NSArray *servicesArray) {
+        result([JSONStringifier jsonStringFromJSONObject:[self arrayReplacingKeys:@[@[@"id", @"serviceId"], @[@"uuid", @"serviceUuid"]] inArray:servicesArray]]);
```
Comment (mikolak) [inline ios/Classes/FlutterBleLibPlugin.m]: break lines after commas (readability), create an issue to refactor `arrayReplacingKeys` if needed

File: ios/Classes/FlutterBleLibPlugin.m [hunk 4]
```diff
@@ -178,9 +210,40 @@ - (void)dispatchEvent:(NSString * _Nonnull)name value:(id _Nonnull)value {
...
+- (Resolve)resolveForCharacteristics:(FlutterResult)result serviceUuid:(NSString *)serviceUuid {
+    return ^(NSArray *servicesArray) {
+
+        NSDictionary *matchingService = nil;
+        for (NSDictionary *service in [self arrayReplacingKeys:@[@[@"id", @"serviceId"], @[@"uuid", @"serviceUuid"]] inArray:servicesArray]) {
+            if ([[service valueForKey:@"serviceUuid"] isEqualToString:serviceUuid]) {
+                matchingService = service;
+                break;
+            }
+        }
```
Comment (mikolak) [inline ios/Classes/FlutterBleLibPlugin.m]: Service might still be nil here.

File: ios/Classes/FlutterBleLibPlugin.m (hunk 1 above)
Comment (TomoLV) [reply ios/Classes/FlutterBleLibPlugin.m]: Removed from this PR

File: ios/Classes/FlutterBleLibPlugin.m (hunk 4 above)
Comment (TomoLV) [reply ios/Classes/FlutterBleLibPlugin.m]: Addressed this issue

File: ios/Classes/FlutterBleLibPlugin.m (hunk 3 above)
Comment (TomoLV) [reply ios/Classes/FlutterBleLibPlugin.m]: Readability improved, issue created

**Pull request merged**